touying-unistra-theme:1.0.0

spidersouris opened the pull request:
<!--
Thanks for submitting a package! Please read and follow the submission guidelines detailed in the repository's README and check the boxes below. Please name your PR as `name:version` of the submitted package.

If you want to make a PR for something other than a package submission, just delete all this and make a plain PR.
-->

I am submitting
- [x] a new package
- [ ] an update for a package

<!--
Please add a brief description of your package below and explain why you think it is useful to others. If this is an update, please briefly say what changed.
-->

touying-unistra-theme is a [Touying](https://github.com/touying-typ/touying) theme for creating presentation slides in [Typst](https://github.com/typst/typst), adhering to the core principles of the [style guide of the University of Strasbourg, France](https://langagevisuel.unistra.fr).

<!--
These things need to be checked for a new submission to be merged. If you're just submitting an update, you can delete the following section.
-->

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

<!--
The following box only needs to be checked for **template** submissions. If you're submitting a package that isn't a template, you can delete the following section. See the guidelines section about licenses in the README for more details.
-->
- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

laurmaedje commented:
Based on the discussion in https://github.com/typst/packages/pull/908, the name will need to incorporate some creative component as `unistra` is canonical and `theme` just descriptive. Since `theme` is somewhat redundant anyway, perhaps one could replace this word with something else?

spidersouris commented:
> Based on the discussion in #908, the name will need to incorporate some creative component as `unistra` is canonical and `theme` just descriptive. Since `theme` is somewhat redundant anyway, perhaps one could replace this word with something else?

Indeed — I've renamed it to "touying-unistra-pristine". Would that work? I've also updated the theme to the latest version of Touying.

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/typst.toml:
```diff
@@ -0,0 +1,21 @@
+license = "MIT"
+description = "Touying theme adhering to the core principles of the style guide of the University of Strasbourg, France"
+# homepage = ""
+repository = "https://github.com/spidersouris/touying-unistra-pristine"
```
This repository does not seem to exist

spidersouris replied:
fixed

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/typst.toml:
```diff
@@ -0,0 +1,21 @@
+compiler = "0.11.0"
+exclude = []
+
+[template]
```
The template will be copied to the user's project and must be self-contained. The relative path referencing the cat will not work as-is. Typically, the directory is also called `template` rather than `example`.

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/example/example.pdf:
The template directory should not contain the compiled PDF. See for example the `charged-ieee` template for how a template should typically be structured.

OrangeX4 commented on packages/preview/touying-unistra-pristine/1.0.0/src/unistra.typ:
```diff
@@ -0,0 +1,854 @@
+#import "@preview/touying:0.5.0": *
```
#import "@preview/touying:0.5.2": *

spidersouris commented:
Thanks for your comments — I have removed the example PDF and changed the template so that it is simpler and does not refer to external images. I have moved the old "template" to the /example dir as has been done by https://github.com/typst/packages/tree/main/packages/preview/touying-buaa/0.1.0 for a more extensive example of how to use the theme.

OrangeX4 commented on packages/preview/touying-unistra-pristine/1.0.0/README.md:
````diff
@@ -0,0 +1,99 @@
+## Import from Typst Universe
+
+```typst
+#import "@preview/touying:0.5.0": *
````
#import "@preview/touying:0.5.2": *

OrangeX4 commented on packages/preview/touying-unistra-pristine/1.0.0/README.md:
````diff
@@ -0,0 +1,99 @@
+See [example/example.typ](example/example.typ) for a complete example with configuration.
+
+```typst
+#import "@preview/touying:0.5.0": *
````
Here.

OrangeX4 commented on packages/preview/touying-unistra-pristine/1.0.0/template/template.typ:
```diff
@@ -0,0 +1,31 @@
+#import "../src/unistra.typ": *
+#import "../src/colors.typ": *
+#import "../src/admonition.typ": *
+#import "../src/settings.typ" as settings
```
我认为这里不能正常使用相对路径？

`#import "@preview/touying-unistra-pristine:1.0.0": *`

OrangeX4 commented on packages/preview/touying-unistra-pristine/1.0.0/example/example.typ:
```diff
@@ -0,0 +1,143 @@
+#import "../src/unistra.typ": *
+#import "../src/colors.typ": *
+#import "../src/admonition.typ": *
+#import "../src/settings.typ" as settings
```
Consider `#import "../lib.typ": *` Instead of importing things separately.

spidersouris replied:
I think importing the theme as it is done in template.typ is the most logical choice as the entry point is src/lib.typ. I have made the necessary changes — thanks!

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/template/template.typ:
```diff
@@ -0,0 +1,28 @@
+#import "@preview/touying:0.5.2": *
+#import "@preview/touying-unistra-pristine:1.0.0": *
+
+#show: unistra-theme.with(
```
This fails with `unknown variable: unistra-theme`. The reason is that the way you've written `lib.typ` with `#import "unistra.typ" as unistra`, it is actually `unistra.unistra-theme`. You probably want to import the specific items you want to be public in `lib.typ` instead.

spidersouris replied:
Indeed — I had the changes in another local folder but forgot to propagate them to the fork. Sorry for the trouble!

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/src/unistra.typ:
```diff
@@ -0,0 +1,856 @@
+#let title-slide(
+  title: "",
+  subtitle: "",
+  logo-path: "../assets/unistra.svg",
```
The logo-path will unfortunately not work with user images because of the way Typst currently resolves path. To fix this, you would need to accept an already loaded `image`, e.g. `logo: image("../assets/unistra.svg")`. Then users can provide their own image. To change the image settings, you can use a a set rule, but the `logo-width` and `logo-height` arguments would just become superfluous.

spidersouris replied:
I have changed the way images are loaded so that:
1) `hero` and `gallery` require inline images to avoid issues with paths,
2) `title-slide` now uses the value of the `logo` parameter in Touying's `config-info` in `show: unistra-theme.with` (it can be overriden)

laurmaedje commented on packages/preview/touying-unistra-pristine/1.0.0/typst.toml:
```diff
@@ -0,0 +1,21 @@
+categories = ["presentation"]
+disciplines = []
+compiler = "0.11.0"
+exclude = []
```
It would be best to exclude the files that are not needed for the package to work. Unless I'm mistaking that would be the `example` folder and the two cat images.

spidersouris replied:
Correct, done

laurmaedje commented:
Thanks!
